catalogs/catalog-lakehouse-paimon/src/test/java/org/apache/gravitino/catalog/lakehouse/paimon/integration/test/CatalogPaimonS3IT.java

```diff
@@ -0,0 +1,140 @@
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class CatalogPaimonS3IT extends CatalogPaimonBaseIT {
+
+  // private static final String S3_BUCKET_NAME =
+  // GravitinoITUtils.genRandomName("paimon-s3-bucket-");
+  private static final String S3_BUCKET_NAME = "my-test-bucket";
+  private static LocalStackContainer localStackContainer;
+  private String accessKey;
+  private String secretKey;
+  private String endpoint;
+
+  private static final String PAIMON_S3_JAR_URL =
+      "https://repo1.maven.org/maven2/org/apache/paimon/paimon-s3/0.8.0/paimon-s3-0.8.0.jar";
+
+  @Override
+  protected Map<String, String> initPaimonCatalogProperties() {
+
+    Map<String, String> catalogProperties = Maps.newHashMap();
+    catalogProperties.put("key1", "val1");
+    catalogProperties.put("key2", "val2");
+
+    TYPE = "filesystem";
+    WAREHOUSE = "s3://" + S3_BUCKET_NAME + "/";
+
+    accessKey = localStackContainer.getAccessKey();
+    secretKey = localStackContainer.getSecretKey();
+    endpoint = localStackContainer.getEndpointOverride(S3).toString();
+
+    catalogProperties.put(PaimonCatalogPropertiesMetadata.GRAVITINO_CATALOG_BACKEND, TYPE);
+    catalogProperties.put(PaimonCatalogPropertiesMetadata.WAREHOUSE, WAREHOUSE);
+    catalogProperties.put(PaimonS3FileSystemConfig.S3_ACCESS_KEY, accessKey);
+    catalogProperties.put(PaimonS3FileSystemConfig.S3_SECRET_KEY, secretKey);
+    catalogProperties.put(PaimonS3FileSystemConfig.S3_ENDPOINT, endpoint);
+
+    // Need to download the S3 dependency in the deploy mode.
+    downloadS3Dependency();
+
+    return catalogProperties;
+  }
+
+  private void downloadS3Dependency() {
+    String gravitinoHome = System.getenv("GRAVITINO_HOME");
+    try {
+      if (!ITUtils.EMBEDDED_TEST_MODE.equals(testMode)) {
+        String serverPath = ITUtils.joinPath(gravitinoHome, "libs");
+        String paimonCatalogPath =
+            ITUtils.joinPath(gravitinoHome, "catalogs", "lakehouse-paimon", "libs");
+        JdbcDriverDownloader.downloadJdbcDriver(PAIMON_S3_JAR_URL, serverPath, paimonCatalogPath);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to download the S3 dependency", e);
+    }
+  }
+
+  @Override
+  protected void startNecessaryContainers() {
+    localStackContainer =
+        new LocalStackContainer(DockerImageName.parse("localstack/localstack")).withServices(S3);
+    localStackContainer.start();
+
+    Awaitility.await()
+        .atMost(60, TimeUnit.SECONDS)
+        .pollInterval(1, TimeUnit.SECONDS)
+        .until(
+            () -> {
+              try {
+                Container.ExecResult result =
+                    localStackContainer.execInContainer(
+                        "awslocal", "s3", "mb", "s3://" + S3_BUCKET_NAME);
+                return result.getExitCode() == 0;
+              } catch (Exception e) {
+                return false;
+              }
+            });
+  }
+
+  @AfterAll
+  public void stop() {
```

Comment:
stop spark session

Reply:
It should be stop in the super stop method, let me check

Reply:
Yeah, the spark session has been closed in the super class.